src/main/java/homework_6/map_problems_generator/Main.java

```diff
@@ -0,0 +1,32 @@
+package homework_6.map_problems_generator;
+
+import java.util.HashMap;
+
+public class Main {
+    public static void main(String[] args) {
```

Comment:
It seems there's no collision, you override the first key-value pair, because your keys will be same for hashcode and equals. Please fix

Reply:
Updated. Generated normal implementation of equals. Collision illustrated by sout in equals.